crates/biome_js_semantic/src/semantic_model/builder.rs

```diff
@@ -19,10 +21,10 @@ pub struct SemanticModelBuilder {
     scope_hoisted_to_by_range: FxHashMap<TextSize, usize>,
     bindings: Vec<SemanticModelBindingData>,
     /// maps a binding range to its index inside SemanticModelBuilder::bindings vec
-    bindings_by_range: FxHashMap<TextRange, usize>,
+    bindings_by_range: FxHashMap<(BindingKind, TextRange), usize>,
     /// maps a reference range to its bindings. usize points to SemanticModelBuilder::bindings vec
-    declared_at_by_range: FxHashMap<TextRange, usize>,
-    exported: FxHashSet<TextRange>,
+    bindings_by_reference_range: FxHashMap<TextRange, usize>,
+    exported: FxHashSet<usize>,
```

Comment:
I can't understand why we don't save the range. What's the reason?

Reply:
Because in some cases we export only the type binding and not the value binding. However, with the [hack I proposed in a previous comment](https://github.com/biomejs/biome/pull/482#discussion_r1347379743), we could use `TextSize`.